server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStatusApisIT.java

```diff
@@ -630,33 +633,61 @@ public void testGetSnapshotsRequest() throws Exception {
     }
 
     public void testConcurrentCreateAndStatusAPICalls() throws Exception {
-        for (int i = 0; i < randomIntBetween(1, 10); i++) {
-            createIndexWithContent("test-idx-" + i);
-        }
+        final var indexNames = IntStream.range(0, between(1, 10)).mapToObj(i -> "test-idx-" + i).toList();
+        indexNames.forEach(this::createIndexWithContent);
         final String repoName = "test-repo";
         createRepository(repoName, "fs");
+
+        if (randomBoolean()) {
+            // sometimes cause some deduplication
+            createSnapshot(repoName, "initial_snapshot", List.of());
+            for (final var indexName : indexNames) {
+                if (randomBoolean()) {
+                    indexDoc(indexName, "another_id", "baz", "quux");
+                }
+            }
+        }
+
         final int snapshots = randomIntBetween(10, 20);
         final List<ActionFuture<SnapshotsStatusResponse>> statuses = new ArrayList<>(snapshots);
         final List<ActionFuture<GetSnapshotsResponse>> gets = new ArrayList<>(snapshots);
         final Client dataNodeClient = dataNodeClient();
-        final String[] snapshotNames = createNSnapshots(repoName, snapshots).toArray(Strings.EMPTY_ARRAY);
 
-        for (int i = 0; i < snapshots; i++) {
+        final var snapshotNames = IntStream.range(0, snapshots).mapToObj(i -> "test-snap-" + i).toArray(String[]::new);
+        final var creates = Arrays.stream(snapshotNames)
+            .map(snapshotName -> clusterAdmin().prepareCreateSnapshot(repoName, snapshotName).execute())
+            .toList();
+        for (final var create : creates) {
+            create.get(10, TimeUnit.SECONDS);
+        }
+
+        var masterClusterService = internalCluster().getCurrentMasterNodeInstance(ClusterService.class);
+        assertBusy(() -> {
+            final var stillRunning = masterClusterService.state()
+                .custom(SnapshotsInProgress.TYPE, SnapshotsInProgress.EMPTY)
+                .isEmpty() == false;
             statuses.add(dataNodeClient.admin().cluster().prepareSnapshotStatus(repoName).setSnapshots(snapshotNames).execute());
```

Comment:
Could we retain the old loop as well? I'm not sure this will trigger the level of concurrency we needed to reproduce the original bug this was supposed to reproduce (the point of this test really is running enough gets and status requests in parallel so that the snapshot meta pool is fully occupied).
Running both loops should be fine IMO, this will only run 16 times at the most over 60s so running another 10-20 extra requests won't hurt IMO.

Reply:
Sure 7ec8e1e1ec471c68a1d8568de627db9b9889cbbd reinstates this (and fixes the number of iterations so it actually does what you said it should :grin:)

Reply:
It strikes me that this bug might also need us to max out the threadpool after the snapshots all completed too - see 99cbdbb.

Reply:
That's a nice idea, I think in practice it's good enough as is but this makes for a lot more action on the meta pool for sure :)